src/handler/updateVin.ts

```diff
@@ -0,0 +1,84 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import 'dotenv/config';
+import { UpdateVinBody } from '../models/updateVin';
+import { setCreatedAuditDetails, setLastUpdatedAuditDetails } from '../services/audit';
+import { getBySystemNumberAndCreatedTimestamp, updateVehicle } from '../services/database';
+import { getUserDetails } from '../services/user';
+import { StatusCode } from '../util/enum';
+import { formatErrorMessage } from '../util/errorMessage';
+import { addHttpHeaders } from '../util/httpHeaders';
+import logger from '../util/logger';
+import { checkStatusCodeValidity } from '../validators/update';
+import { validateAmendVinPayloadErrors } from '../validators/updateVin';
+import { checkVinValidity } from '../validators/vinValidity';
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  try {
+    logger.debug('Amend VIN Called');
+
+    const amendVinPayloadErrors = validateAmendVinPayloadErrors(event);
+
+    if (amendVinPayloadErrors) {
+      return addHttpHeaders(amendVinPayloadErrors);
+    }
+
+    const userDetails = getUserDetails(event.headers.Authorization ?? '');
+    const systemNumber = decodeURIComponent(event.pathParameters?.systemNumber ?? '');
+    const createdTimestamp = decodeURIComponent(event.pathParameters?.createdTimestamp ?? '');
+    const body = await JSON.parse(event.body ?? '') as UpdateVinBody;
+
+    const recordFromDB = await getBySystemNumberAndCreatedTimestamp(systemNumber, createdTimestamp);
+
+    const statusCodeErrors = checkStatusCodeValidity(recordFromDB.techRecord_statusCode);
+    if (statusCodeErrors) {
+      return addHttpHeaders(statusCodeErrors);
+    }
+
+    const vinValidityErrors = checkVinValidity(recordFromDB.vin, body.newVin);
+    if (vinValidityErrors) {
+      return addHttpHeaders(vinValidityErrors);
+    }
+
+    let partialVin;
+    if (body.newVin.length < 6) {
+      partialVin = body.newVin.toUpperCase();
+    } else {
+      partialVin = body.newVin.substring(Math.max(body.newVin.length - 6)).toUpperCase();
+    }
```

Comment:
could do something like this?

    const partialVin = extractPartialVin(body.newVin);